components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -237,4 +299,17 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
     // Drop the previous fragment and add the incoming fragment.
     this.router.navigate([this.url.split('#')[0]], { fragment: event.target.value });
   }
+
+  onProjectChecked(project: ProjectChecked): void {
+    this.projects[project.id] = project;
+  }
+
+  noProjectsUpdated(): boolean {
+    return xor(this.team.projects,
```

Comment:
how does `xor` work?

Reply:
In this case, [xor](https://en.wikipedia.org/wiki/Exclusive_or) finds all projects that were in only one or the other of the two lists. So if either list contains a team that the other doesn't we'll know that there are projects to save.